src/main/java/essenmakanan/ingredient/IngredientList.java

```diff
@@ -74,7 +74,31 @@ public int getIndex(String ingredientName) {
     }
 
     public void addIngredient(Ingredient ingredient) {
+        if (this.exist(ingredient.getName())) {
+            try{
+                this.updateIngredient(ingredient);
+            } catch (EssenFormatException e) {
+                e.handleException();
+            }
+            return;
+        }
         ingredients.add(ingredient);
+        Ui.printAddIngredientsSuccess(ingredient.getName());
```

Comment:
Will it be better to put this under else? So that you dont have to use return; and maybe it would be clearer that you are handling 2 different cases.

Reply:
good point